woocommerce/woocommerce
# Update `@woocommerce/data` & admin client type definitions

### All Submissions:






### Changes proposed in this Pull Request:



Closes #32692.

This PR adds the `@types/wordpress__data` and `@types/wordpress__compose`, `redux` types as `@woocommerce/data` devDependencies, and fixes the consequent error.

- [Fix @woocommerce/data type errors after enabling wp data, compose types](https://github.com/woocommerce/woocommerce/pull/32735/commits/5c8f76bbe78db1a219bbb59ac24f90b90a08c771)
- [Fix and export @woocommerce/data types for react admin ./client](https://github.com/woocommerce/woocommerce/pull/32735/commits/672ba17a807962d7eed0fb2597f3dc398d94d5cb)
- [Fix react admin ./client types after updating @woocommerce/data types](https://github.com/woocommerce/woocommerce/pull/32735/commits/678e893f60845385e9b0cbe245379aa29e6e7590)

Many type errors appear because `@types/wordpress__data` is added, making TS able to check related types. This is also why we don't need to manually specify the wp `select` type (https://github.com/woocommerce/woocommerce/pull/32735/commits/d94c1dc7bd0a7c6c7512dda3e4abed9b320ea3b8).

I referred to [wp-calypso](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/site/index.ts#L16-L33) to find out how to add redux store types.


### How to test the changes in this Pull Request:

1. Run `pnpm install`
2. Run `pnpm nx build @woocommerce/data`, confirm no errors
3. Run `pnpm nx ts:check woocommerce-admin`, confirm no errors

### Other information:




### FOR PR REVIEWER ONLY:
